src/huggingface_hub/utils/_http.py

```diff
@@ -590,3 +591,41 @@ def _curlify(request: requests.PreparedRequest) -> str:
             flat_parts.append(quote(v))
 
     return " ".join(flat_parts)
+
+
+def _adjust_range_header(original_range: Optional[str], resume_size: int) -> Optional[str]:
+    """
+    Adjust HTTP Range header to account for resume position.
+    """
+    if not original_range:
+        return f"bytes={resume_size}-"
+
+    if "," in original_range:
+        warnings.warn(f"Multiple ranges detected - {original_range!r}, using full range after resume", UserWarning)
+        return f"bytes={resume_size}-"
+    # Regex to parse HTTP Range header
+    range_regex = re.compile(r"^\s*bytes\s*=\s*(\d*)\s*-\s*(\d*)\s*$", re.IGNORECASE)
+
+    match = range_regex.match(original_range)
+    if not match:
+        raise RuntimeError(f"Invalid range format - {original_range!r}.")
+    start, end = match.groups()
+
+    if not start:
+        if not end:
+            raise RuntimeError(f"Invalid range format - {original_range!r}.")
+
+        new_suffix = int(end) - resume_size
+        if new_suffix <= 0:
+            return None
+        return f"bytes=-{new_suffix}"
+
+    start = int(start)
+    new_start = start + resume_size
+    if end:
+        end = int(end)
+        if new_start > end:
+            return None
```

Comment:
Shouldn't we raise instead since it's an unsatisfiable range? In which case we would like to ignore the requested range?

Reply:
The same as the above, `new_start > end` means download completed.

Reply:
If it's equal, then yes I'd say it means it's completed downloaded. But I don't see a case where it'd be `>` and not be an invalid state. At the moment, if we return `None`, the full file is downloaded as per https://github.com/huggingface/huggingface_hub/pull/2823/files#diff-765507f4602931b08aa20d0177611d5da7d5c7805882b0a79c81b5ed72c9472aR369